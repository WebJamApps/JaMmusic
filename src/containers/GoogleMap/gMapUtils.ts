import type { Iloc } from '.';
import type UMap from './UserMap';
import type CMap from './CompanyMap';

function makeLoc(faker: { address: { latitude: () => string; longitude: () => string; }; }): Iloc {
  return {
    lat: parseFloat(faker.address.latitude()),
    lng: parseFloat(faker.address.longitude()),
  };
}
const limitLat = (obj: UMap | CMap):UMap | CMap => {
  const newObj = obj;
  if (obj.loc.lat > 83) newObj.loc.lat = 83;
  if (obj.loc.lat < -70) newObj.loc.lat = -70;
  return newObj;
};
function limitUserLat(obj: UMap): UMap { return limitLat(obj); }

function limitCompanyLat(obj: CMap): CMap { return limitLat(obj); }

export default { makeLoc, limitUserLat, limitCompanyLat };
