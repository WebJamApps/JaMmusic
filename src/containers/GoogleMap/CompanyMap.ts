import faker from 'faker';
import type { Iloc } from '.';
import gMapUtils from './gMapUtils';

export class CompanyMap {
  name: string;

  catchPhrase?: string;

  loc: Iloc;

  constructor() {
    this.name = faker.company.companyName();
    this.catchPhrase = faker.company.catchPhrase();
    this.loc = gMapUtils.makeLoc(faker);
  }
}

export default CompanyMap;
