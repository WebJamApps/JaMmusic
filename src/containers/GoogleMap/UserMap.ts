import faker from 'faker';
import type { Iloc } from '.';
import gMapUtils from './gMapUtils';

export class UserMap {
  name: string;

  loc: Iloc;

  constructor() {
    this.name = faker.name.firstName();
    this.loc = gMapUtils.makeLoc(faker);
  }
}

export default UserMap;
