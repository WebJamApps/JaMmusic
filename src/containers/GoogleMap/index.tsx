
import { Component } from 'react';
import UMap from './UserMap';
import CMap from './CompanyMap';
import gMapUtils from './gMapUtils';

export interface Iloc { lat: number, lng: number }
export const equals = () => true;
class GoogleMap extends Component {
  gMap: google.maps.Map | null;

  companyMap: CMap | null;

  userMap: UMap | null;

  content: string;

  gMapUtils: typeof gMapUtils;

  constructor(props: Record<string, unknown>) {
    super(props);
    this.gMap = null;
    this.userMap = null;
    this.companyMap = null;
    this.content = '';
    this.gMapUtils = gMapUtils;
  }

  componentDidMount(): void {
    const mapDiv = document.getElementById('googleMap');
    if (mapDiv) this.gMap = new google.maps.Map(mapDiv, { zoom: 2, center: { lat: 40, lng: -100 } });
    this.userMap = new UMap();
    this.userMap = this.gMapUtils.limitUserLat(this.userMap);
    this.content = `<div><p><strong>User Name:</strong> ${this.userMap.name}</p>`
      + `<p><strong>Latitude:</strong> ${this.userMap.loc.lat}</p></div>`;
    this.addMarker(this.userMap);
    this.companyMap = new CMap();
    this.companyMap = this.gMapUtils.limitCompanyLat(this.companyMap);
    this.content = `<div><p><strong>Company Name:</strong> ${this.companyMap.name}</p>`
      + `<p><strong>Company Slogan:</strong> ${this.companyMap.catchPhrase}</p>`
      + `<p><strong>Latitude:</strong> ${this.companyMap.loc.lat}</p></div>`;
    this.addMarker(this.companyMap, {
      scaledSize: { width: 40, height: 40, equals },
      url: 'https://image.flaticon.com/icons/png/512/63/63838.png',
    });
  }

  addMarker(obj: { loc: Iloc }, icon?: google.maps.Icon | undefined): void {
    const infoWindow = new google.maps.InfoWindow({ content: this.content });
    if (this.gMap !== null) {
      const marker = new google.maps.Marker({
        map: this.gMap,
        position: {
          lat: obj.loc.lat, lng: obj.loc.lng,
        },
        icon,
      });
      marker.addListener('click', () => { // eslint-disable-next-line security/detect-non-literal-fs-filename
        /* istanbul ignore else */if (this.gMap !== null)infoWindow.open(this.gMap, marker);
      });
    }
  }

  render(): JSX.Element {
    return (
      <div className="page-content google">
        <div style={{ margin: 'auto', textAlign: 'center' }}>
          <h3 style={{ marginTop: '10px' }}>Google Map Here</h3>
          <div id="googleMap" style={{ height: '8.5in', maxWidth: '11in', margin: 'auto' }} />
        </div>
      </div>
    );
  }
}
export default GoogleMap;
