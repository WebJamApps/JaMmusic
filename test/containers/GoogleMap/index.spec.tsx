/* eslint-disable max-classes-per-file */
import renderer from 'react-test-renderer';
import GoogleMap, { equals } from 'src/containers/GoogleMap';

describe('GoogleMap container', () => {
  let lType = '';
  const fakeFunc: any = jest.fn();
  class Marker { addListener(type: string, cb: () => void) { lType = type; cb(); } }
  class InfoWindow { open() { } }
  const fakeMarker: any = Marker;
  const fakeInfoWindow: any = InfoWindow;
  beforeEach(() => {
    global.google = {
      maps: {
        ...window.google.maps,
        Marker: fakeMarker,
        Map: fakeFunc,
        InfoWindow: fakeInfoWindow,
      },
    };
  });
  it('renders correctly', () => {
    const gMap = renderer.create(<GoogleMap />).toJSON();
    expect(gMap).toMatchSnapshot();
    expect(lType).toBe('click');
  });
  it('equals is true', () => {
    expect(equals()).toBe(true);
  });
});
