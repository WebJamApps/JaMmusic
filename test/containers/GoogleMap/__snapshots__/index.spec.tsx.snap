// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleMap container renders correctly 1`] = `
<div
  className="page-content google"
>
  <div
    style={
      {
        "margin": "auto",
        "textAlign": "center",
      }
    }
  >
    <h3
      style={
        {
          "marginTop": "10px",
        }
      }
    >
      Google Map Here
    </h3>
    <div
      id="googleMap"
      style={
        {
          "height": "8.5in",
          "margin": "auto",
          "maxWidth": "11in",
        }
      }
    />
  </div>
</div>
`;
